class_15_sockets/homework/server.py

```diff
@@ -0,0 +1,78 @@
+import pathlib
+import selectors
+import socket
+from collections import defaultdict
+from multiprocessing import Lock
+from multiprocessing.pool import ThreadPool
+
+HOST = "localhost"
+PORT = 9999
+
+selector = selectors.DefaultSelector()
+pool = ThreadPool(4)
+bufs = defaultdict(bytes)
+clients = defaultdict(str)
+lock = Lock()
+
+
+def append_data(data):
+    data = data.decode()
+    pathlib.Path("outputs").mkdir(parents=True, exist_ok=True)
+    with lock:
+        print("writing to a file")
+        with open(f"outputs/output.txt", "a", encoding="utf-8") as f:
+            f.write(data)
```

Comment:
May be combine this lines

Reply:
Alright,
I'll combine writing data and a newline as well as using `with` lock and `open` function